pkg/util/fieldindex/register.go

```diff
@@ -152,3 +163,35 @@ func indexImagePullJobActive(c cache.Cache) error {
 		return []string{isActive}
 	})
 }
+
+func indexSideCatSet(c cache.Cache) error {
+	return c.IndexField(context.TODO(), &appsv1alpha1.SidecarSet{}, IndexNamespace, func(rawObj client.Object) []string {
+		obj := rawObj.(*appsv1alpha1.SidecarSet)
+		if obj == nil {
+			return nil
+		}
+
+		ns := sets.NewString()
```

Comment:
I think there are two types, one is configured spec.namespace and the other is not configured which you can set 'CludsterScope' Namespace, and namespaceSelector also counts as a type of 'ClusterScope'.